R/check-class.R

```diff
@@ -102,17 +108,44 @@ check_update <- function(self, private) {
 
 check_web <- function(self, private, which) {
 
+  ids <- select_ids(which = which, self = self, 
+                    private = private)
+
+  urls <- paste0(sub("/api$", "/status/", baseurl()), ids)
+  
+  lapply(urls, browseURL)
+  invisible(self)
+}
+
+check_urls <- function(self, private, which) {
+  
+  ids <- select_ids(which = which, self = self, 
+                    private = private)
+  
+  urls <- paste0(sub("/api$", "/status/", baseurl()), ids)
+  
+  do.call("rbind",
+          lapply(ids, get_check_urls))
+}
+
+get_check_urls <- function(id){
+  data.frame(html = paste0(sub("/api$", "/status/", baseurl()), id),
+             text = paste0(sub("/api$", "/status/original/", baseurl()), id),
+             artifacts = paste0("https://artifacts.r-hub.io/", id),
+             stringsAsFactors = FALSE)
+}
+
+select_ids <- function(which, self, private){
   ids <- if (is.null(which)) {
     private$ids_
   } else if (is.numeric(which)) {
     private$ids_[which]
   } else if (is.character(which)) {
     intersect(private$ids_, which)
   } else {
-    stop("Unknown check selected")
+    stop("Unknown check selected",
+         call. = FALSE)
   }
-
-  urls <- paste0(sub("/api$", "/status/", baseurl()), ids)
-  lapply(urls, browseURL)
-  invisible(self)
-}
+  
+  return(ids)
+}
```

Comment:
You need a newline at the end of the file.

Reply:
:heavy_check_mark: